examples/barotropic_instability.jl

```diff
@@ -0,0 +1,288 @@
+    t₀ = simulation.model.clock.time = 0
+    e₀ = e[1, 1, 1]
+
+    # Grow
+    run!(simulation)
+
+    # Analyze
+    compute!(e)
+    e₁ = e[1, 1, 1]
+    Δt = simulation.model.clock.time - t₀
+
+    # (u² + v²) / 2 ~ exp(2 σ Δt)
+    σ = growth_rate = log(e₁ / e₀) / 2Δt
+
+    return growth_rate    
+end
+
+# Finally, we write a function that rescales the velocity field
+# at thewrite 
+
+"""
+    rescale!(velocities; factor=1e-1)
+
+"""
+function rescale!(velocities; factor=1e-1)
+    velocities.u.data.parent .*= factor
+    velocities.v.data.parent .*= factor
+    velocities.w.data.parent .*= factor
+    return nothing
+end
+
+using Printf
+
+relative_change(σⁿ, σⁿ⁻¹) = isfinite(σⁿ) ? abs((σⁿ - σⁿ⁻¹) / σⁿ) : Inf
+
+"""
+    estimate_growth_rate!(simulation, e, ω; convergence_criterion=1e-2)
+
+Computes 
+"""
+function estimate_growth_rate!(simulation, e, ω; convergence_criterion=1e-1)
+    σ = [0.0, grow_instability!(simulation, e)]
+
+    while relative_change(σ[end], σ[end-1]) > convergence_criterion
+
+        push!(σ, grow_instability!(simulation, e))
+
+        @info @sprintf("*** Power iteration %d, σⁿ: %.2e, relative Δσ: %.2e",
+                       length(σ), σ[end], relative_change(σ[end], σ[end-1]))
+
+        compute!(ω)
+        display(eigenplot!(interior(ω)[:, :, 1], σ, nothing))
+
+        rescale!(simulation.model.velocities)
+    end
+
+    return σ
+end
+
+# # Eigenplotting
+#
+# A good algorithm wouldn't be complete without a good visualization tool,
+
+using Oceananigans.AbstractOperations
+
+u, v, w = model.velocities
+
+vorticity = ComputedField(∂x(v) - ∂y(u))
+
+x, y, z = nodes(vorticity)
+
+eigentitle(σ, t) = "Iteration $(length(σ)): most unstable eigenmode"
+eigentitle(::Nothing, t) = @sprintf("Vorticity at t = %.2f", t)
+
+function eigenplot!(ω, σ, t; ω_lim=maximum(abs, ω)+1e-16)
+    
+    background_flow = plot(bickley_jet.(0, y, 0, 0), y, label=nothing,
+                           ylims=(grid.yF[1], grid.yF[grid.Ny]), xlabel="U(y)", ylabel="y")
+
+    ω_contours = contourf(x, y, clamp.(ω, -ω_lim, ω_lim)';
+                          color = :balance, aspectratio = 1,
+                          levels = range(-ω_lim, stop=ω_lim, length=21),
+                          xlims = (grid.xF[1], grid.xF[grid.Nx]),
+                          ylims = (grid.yF[1], grid.yF[grid.Ny]),
+                          clims = (-ω_lim, ω_lim), linewidth = 0)
+                          
+    eigenplot = plot(background_flow, ω_contours,
+                     layout = Plots.grid(1, 2, widths=[0.2, 0.8]),
+                     link = :y, size = (600, 200),
+                     title = ["Background flow" eigentitle(σ, t)])
+
+    return eigenplot
+end
+
+# # Rev your engines...
+#
+# We initialize the power iteration with random noise.
+# The amplitude of the initial condition is arbitrary since our algorithm
+# will rescale the velocity field iteratively until the simulation's stop_criteria
+# is no longer met.
```

Comment:
The amplitude is, indeed, arbitrary. But if you wanna find the eigenmode you have to make sure that your "perturbations" are small-enough amplitude so that you are in the linear regime. I'll add a remark...